pkg/operator/encryption/deployer/revisionedpod.go

```diff
@@ -141,7 +141,13 @@ func (d *RevisionLabelPodDeployer) HasSynced() bool {
 //   1. All running pods are ready and at the same revision
 //   2. All master nodes have a running pod
 //   3. There are no pending or unknown pods
-//   4. All succeeded and failed pods have revisions that are before the running pods
+//   4. We tolerate pods in Succeeded or Failed phase only if their revision <= the running pods
+//      It turns out that when a node dies or is disconnected from the rest of the cluster, Kubernetes (kubelet) applies a policy for setting the phase of all Pods on the lost node to Failed.
+//      Pods in the Failed state cannot transition back to Running state.
+//      In addition, failed Pods are note GC'ed. The API objects remain in the cluster's API until a human or controller process explicitly removes them.
+//      See the following for more:
+//        https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/
+//        https://bugzilla.redhat.com/show_bug.cgi?id=2000276
```

Comment:
the BZ is about static pods. The pod life-cycle is about non-static pods, isn't it?

Reply:
the title is wrong then. The description points that there were `pods` in `InvalidNodeInfo` state in the `openshift-apiserver` ns